watertap/tools/parameter_sweep.py

```diff
@@ -319,108 +338,239 @@ def _interp_nan_values(global_values, global_results):
-                                       default IDAES solver, raising an exception if the termination
-                                       condition is not optimal.
+def _create_component_output_skeleton(component, num_samples):
+
+    comp_dict = {}
+    comp_dict["value"] = np.zeros(num_samples, dtype=np.float)
+    if hasattr(component, 'lb'):
+        comp_dict["lower bound"] = component.lb
+    if hasattr(component, 'ub'):
+        comp_dict["upper bound"] = component.lb
+    if hasattr(component, 'get_units'):
+        unit_obj = component.get_units()
+        if unit_obj is not None:
+            comp_dict["units"] = component.get_units().name
+        else:
+            comp_dict["units"] = "non-dimensional"
 
-        optimize_kwargs (optional) : Dictionary of kwargs to pass into every call to
-                                     ``optimize_function``. The first arg will always be ``model``,
-                                     e.g., ``optimize_function(model, **optimize_kwargs)``. The default
-                                     uses no kwargs.
+    return comp_dict
 
-        reinitialize_function (optional) : A user-defined function to perform the re-initialize the 
-                                           flowsheet ``model`` if the first call to ``optimize_function``
-                                           fails for any reason. After ``reinitialize_function``, the
-                                           parameter sweep tool will immediately call
-                                           ``optimize_function`` again.
+# ================================================================
 
-        reinitialize_kwargs (optional) : Dictionary or kwargs to pass into every call to 
-                                         ``reinitialize_function``. The first arg will always be
-                                         ``model``, e.g.,
-                                         ``reinitialize_function(model, **reinitialize_kwargs)``.
-                                         The default uses no kwargs.
+def _update_local_output_dict(model, sweep_params, case_number, sweep_vals,
+        solver_termination_condition, output_dict):
 
-        mpi_comm (optional) : User-provided MPI communicator for parallel parameter sweeps.
-                              If None COMM_WORLD will be used. The default is sufficient for most
-                              users.
+    # Get the inputs
+    op_ps_dict = output_dict["sweep_params"]
+    for key, item in sweep_params.items():
+        var_name = item.pyomo_object.name
+        op_ps_dict[var_name]['value'][case_number] = item.pyomo_object.value
 
-        debugging_data_dir (optional) : Save results on a per-process basis for parallel debugging
-                                        purposes. If None no `debugging` data will be saved.
+    # Get the outputs from model
+    if solver_termination_condition == "optimal":
```

Comment:
Why not use `pyo.check_optimal_termination_condition`?

Reply:
I have updated it to use `pyo.check_optimal_termination` in the commit 878aa3a1a74aaf9b79ef1ac65249bf0591801d0a. I agree that its probably a better way to deal with optimal solutions, although, looking at its definition, I don't think the values will change when using other optimizers besides IPOPT.

https://github.com/Pyomo/pyomo/blob/fa26176776e0d3f4f341d8b93c13ae554f3e8da2/pyomo/opt/results/solver.py#L129-L147

Reply:
@k1nshuk Not all solvers return "optimal" as their final state given a successful solution. For example, some solvers will return "locally_optimal" instead - it has the same general meaning but would fail the `assert`.

Reply:
Right, I was just trying to say that Pyomo seems to be standardizing the optimal termination conditions to `optimal`, `locallyOptimal`, and `globallyOptimal`. I do agree that simply checking for "optimal" (what I had there) is not enough.